registrations/tests/test_signup_group_patch.py

```diff
@@ -33,202 +41,312 @@ def assert_patch_signup_group(api_client, signup_group_pk, signup_group_data):
 
 
+@pytest.mark.parametrize("admin_type", ["superuser", "registration_admin"])
 @freeze_time("2023-03-14 03:30:00+02:00")
 @pytest.mark.django_db
-def test_registration_admin_can_patch_signups_data(user_api_client, registration, user):
-    user.get_default_organization().registration_admin_users.add(user)
+def test_registration_user_who_is_superuser_or_registration_admin_can_patch_signups_data(
+    api_client, registration, admin_type
+):
+    user = UserFactory(is_superuser=True if admin_type == "superuser" else False)
+    if admin_type == "registration_admin":
+        user.registration_admin_organizations.add(registration.publisher)
+
+    RegistrationUserAccessFactory(registration=registration, email=user.email)
 
     signup_group = SignUpGroupFactory(registration=registration)
-    signup0 = SignUpFactory(signup_group=signup_group, registration=registration)
-    signup1 = SignUpFactory(signup_group=signup_group, registration=registration)
+    first_signup = SignUpFactory(signup_group=signup_group, registration=registration)
+    second_signup = SignUpFactory(signup_group=signup_group, registration=registration)
 
     signup_group_data = {
         "signups": [
-            {"id": signup0.pk, "presence_status": SignUp.PresenceStatus.PRESENT},
-            {"id": signup1.pk, "extra_info": "signup1 extra info"},
+            {"id": first_signup.pk, "presence_status": SignUp.PresenceStatus.PRESENT},
+            {"id": second_signup.pk, "extra_info": "signup2 extra info"},
         ]
     }
 
-    assert signup0.presence_status == SignUp.PresenceStatus.NOT_PRESENT
-    assert signup1.presence_status == SignUp.PresenceStatus.NOT_PRESENT
-    assert signup0.extra_info is None
-    assert signup1.extra_info is None
+    assert first_signup.presence_status == SignUp.PresenceStatus.NOT_PRESENT
+    assert second_signup.presence_status == SignUp.PresenceStatus.NOT_PRESENT
+    assert first_signup.extra_info is None
+    assert second_signup.extra_info is None
+
+    api_client.force_authenticate(user)
 
-    assert_patch_signup_group(user_api_client, signup_group.id, signup_group_data)
+    assert_patch_signup_group(api_client, signup_group.id, signup_group_data)
 
-    signup0.refresh_from_db()
-    signup1.refresh_from_db()
-    assert signup0.presence_status == SignUp.PresenceStatus.PRESENT
-    assert signup1.presence_status == SignUp.PresenceStatus.NOT_PRESENT
-    assert signup0.extra_info is None
-    assert signup1.extra_info == "signup1 extra info"
+    first_signup.refresh_from_db()
+    second_signup.refresh_from_db()
+    assert first_signup.presence_status == SignUp.PresenceStatus.PRESENT
+    assert second_signup.presence_status == SignUp.PresenceStatus.NOT_PRESENT
+    assert first_signup.extra_info is None
+    assert second_signup.extra_info == "signup2 extra info"
 
 
 @freeze_time("2023-03-14 03:30:00+02:00")
 @pytest.mark.django_db
 def test_admin_cannot_patch_signups_data(user_api_client, registration):
```

Comment:
`test_admin_cannot_patch_signups_data` and `test_regular_user_cannot_patch_signups_data` are quite similar. Maybe there could be only one test case and use parameter to define user role

```
@pytest.mark.parametrize(
    "user_role",
    [
        "admin",
        "regular",
    ],
)
@freeze_time("2023-03-14 03:30:00+02:00")
@pytest.mark.django_db
def test_admin_or_regular_user_cannot_patch_signups_data(
    api_client, event, registration, user_role
):
    if user_role == "admin":
        user = UserFactory()
        user.admin_organizations.add(event.publisher)
    elif user_role == "regular":
        user = UserFactory()
        user.organization_memberships.add(event.publisher)

    signup_group = SignUpGroupFactory(registration=registration)
    first_signup = SignUpFactory(signup_group=signup_group, registration=registration)
    second_signup = SignUpFactory(signup_group=signup_group, registration=registration)

    signup_group_data = {
        "signups": [
            {"id": first_signup.pk, "presence_status": SignUp.PresenceStatus.PRESENT},
            {"id": second_signup.pk, "extra_info": "signup2 extra info"},
        ]
    }

    assert first_signup.presence_status == SignUp.PresenceStatus.NOT_PRESENT
    assert second_signup.presence_status == SignUp.PresenceStatus.NOT_PRESENT
    assert first_signup.extra_info is None
    assert second_signup.extra_info is None

    api_client.force_authenticate(user)
    response = patch_signup_group(api_client, signup_group.id, signup_group_data)
    assert response.status_code == status.HTTP_403_FORBIDDEN

    first_signup.refresh_from_db()
    second_signup.refresh_from_db()
    assert first_signup.presence_status == SignUp.PresenceStatus.NOT_PRESENT
    assert second_signup.presence_status == SignUp.PresenceStatus.NOT_PRESENT
    assert first_signup.extra_info is None
    assert second_signup.extra_info is None
```